beacon-chain/forkchoice/doubly-linked-tree/epbs_test.go

```diff
@@ -79,6 +81,117 @@ func TestStore_Insert_PayloadContent(t *testing.T) {
 	require.Equal(t, n, n2)
 }
 
+func TestStore_Insert_PayloadEnvelope(t *testing.T) {
```

Comment:
<img width="706" alt="Screenshot 2024-09-06 at 5 58 49 PM" src="https://github.com/user-attachments/assets/9eed53d0-bf4f-4256-9413-6cb24579d825">

Attaching my memo of a tree of nodes in the hope that it might help future reviewers to follow. Adopting consistent naming distinguishing empty node and full node and not reusing `n` could enhance readability.